File: src/main/java/com/aventstack/klov/viewdefs/Color.java
package com.aventstack.klov.viewdefs;

public class Color {

    public Color() { }
    
    public String byStatus(String status) {
        switch (status.trim().toLowerCase()) {
            case "pass" : return "green";
            case "fail" : return "red";
            case "fatal" : return "red";
            case "error" : return "amber";
            case "warning" : return "yellow";
            case "skip" : return "yellow";
            case "info" : return "light-blue";
            case "debug" : return "";
            default:
                return "black";
        }
    }
    
    public String getBootstrapKeyword(String status) {
        switch (status.trim().toLowerCase()) {
            case "pass" : return "success";
            case "fail" : return "danger";
            case "fatal" : return "danger";
            case "error" : return "warning";
            case "warning" : return "warning";
            case "skip" : return "primary";
            case "info" : return "info";
            case "debug" : return "";
            default:
                return "black";
        }
    }
    
}

File: src/main/resources/static/api/factory/Color.js
angular.module('Klov').
factory('Color', function() {
    return {
        getColor: function(status) {
            switch (status) {
                case 'pass': return('green');
                case 'fail': return('red');                
                case 'fatal': return('red');
                case 'error': return('amber');
                case 'warning': return('yellow');
                case 'skip': return('yellow');                
                case 'info': return('light-blue');
                default:
                    return('');
            };
        },
        
        getBootstrapColor: function(status) {
        	switch (status) {
	            case 'pass': return('success');
	            case 'fail': return('danger');                
	            case 'fatal': return('danger');
	            case 'error': return('danger');
	            case 'warning': return('warning');
	            case 'skip': return('warning');                
	            case 'info': return('info');
	            default:
	                return('');
	        };
        }
    }
});
